### PR TITLE
Craftable material sniffers

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Are YOU annoyed about never finding the fucking mats you need? Slap some cable coil on an analyzer and find that untouched sheet of iron and or glass. It's a pinpointer for basic sheets.

## Why It's Good For The Game

Originally I meant to give this as something borgs might want, but I realized they really won't have a great time using it and the situation is more about botanists not being asked, and not knowing how to grow lots of iron, repeat for all other departments. Awareness problems!!!

But I still like the item as a miscellaneous craftable.

## Changelog
:cl:
add: Craftable Material sniffers
/:cl:

### CODE REVIEW

File: code/game/objects/items/stacks/sheets/sheets.dm [hunk 1]
```diff
@@ -17,11 +17,20 @@
...
 	. = ..()
 	pixel_x = rand(-4, 4)
 	pixel_y = rand(-4, 4)
+	if(sniffable)
+		GLOB.sniffable_sheets += src
+
+/obj/item/stack/sheet/Destroy(force)
+	if(sniffable)
+		GLOB.sniffable_sheets -= src
+	. = ..()
```
Comment (Fikou) [inline code/game/objects/items/stacks/sheets/sheets.dm:33]:
```suggestion
	return ..()
```

**Comment by Sealed101** [comment]
this but for materials
image

**Comment by optimumtact** [comment]
this should spawn a holographic dog that runs towards the metal

**Comment by tralezab** [comment]
> this but for materials 

let. him. cook.

File: code/datums/components/crafting/equipment.dm [hunk 2]
```diff
@@ -142,6 +142,17 @@
 	result = /obj/item/ore_sensor
 	category = CAT_EQUIPMENT
 
+/datum/crafting_recipe/material_sniffer
+	name = "Material Sniffer"
+	time = 3 SECONDS
+	reqs = list(
+		/obj/item/analyzer = 1,
```
Comment (ChungusGamer666) [inline code/datums/components/crafting/equipment.dm:149]: but why is it an analyzer and not idk, a gps?

File: code/game/objects/items/stacks/sheets/sheets.dm [hunk 3]
```diff
@@ -17,11 +17,27 @@
...
+		GLOB.sniffable_sheets += src
+
+/obj/item/stack/sheet/Destroy(force)
+	if(sniffable)
+		GLOB.sniffable_sheets -= src
+	return ..()
+
+/obj/item/stack/sheet/add(_amount)
+	. = ..()
+	if(sniffable && is_station_level(z) && amount < 10)
```
Comment (Hatterhat) [inline code/game/objects/items/stacks/sheets/sheets.dm:37]: this amount check looks a lil funky

Comment (tralezab) [reply code/game/objects/items/stacks/sheets/sheets.dm:37]: you might be very right

**Review by Imaginos16 — APPROVED** [review]
The sprite is okay for now.

**Review by ArcaneMusic — APPROVED** [review]
Code looks clean

**Pull request merged**